pilot/pkg/serviceregistry/kube/controller/fake.go

```diff
@@ -220,8 +221,11 @@ func NewFakeControllerWithOptions(t test.Failer, opts FakeControllerOptions) (*F
 	if x, ok := xdsUpdater.(*FakeXdsUpdater); ok {
 		fx = x
 	}
-	go c.Run(c.stop)
-	kubelib.WaitForCacheSync(c.stop, c.HasSynced)
+
+	if !opts.SkipRun {
+		go c.Run(c.stop)
```

Comment:
I feel the right fix is to split out `run` from `new`.

Reply:
The issue is that then each test needs to remember to call it, and needs to know which stop channel to call it with, close the stop channel, etc. It's not a huge amount of work but adds up.

We could have 2 methods (one that runs and one doesn't).but figured using Options was easier since we already have it and SkipRun is used in other places

Reply:
Yes, this is quick and i am concerned in future use, one could easily forget to set the option. Whatever I am good with it right now.